fixtures/dom/src/components/fixtures/number-inputs/NumberTestCaseShadowDOM.js

```diff
@@ -0,0 +1,66 @@
+import Fixture from '../../Fixture';
+
+const React = window.React;
+
+class NumberTestCaseShadowDOM extends React.Component {
+  state = {value: ''};
+
+  controlledInputElementText = `<input
+    type="number"
+    value={this.state.value}
+    onChange={this.onChange}
+  />`;
+
+  uncontrolledInputElementText = `<input type="number" defaultValue={0.5} />`;
+
+  onChange = event => {
+    const parsed = parseFloat(event.target.value, 10);
+    const value = isNaN(parsed) ? '' : parsed;
+
+    this.setState({value});
+  };
+
+  componentDidMount() {
+    this.attachShadowDomElement(
+      '#shadow-dom-host-controlled',
+      this.controlledInputElementText
+    );
+    this.attachShadowDomElement(
+      '#shadow-dom-host-uncontrolled',
+      this.uncontrolledInputElementText
+    );
+  }
+
+  attachShadowDomElement(hostElementId, stringHtmlElement) {
+    var hostElement = document.querySelector(hostElementId);
+    var shadow = hostElement.attachShadow({mode: 'open'});
+    shadow.innerHTML = stringHtmlElement;
```

Comment:
It looks like this injects`<input type="number" value={this.state.value} onChange={this.onChange} />`, which is the form React requires, however this won't wire on `onChange`, just set an HTML string uncontrolled by React. This will create a false positive for the bug because React doesn't interact with this injected markup.

We need to render a React component inside of a shadow dom sub-tree. It looks like https://github.com/facebook/react/issues/11827 includes a reproduction that does just this:
https://jsfiddle.net/69z2wepo/94566/

Drawing from that example, you could do:

```javascript
import NumberTestCase from './NumberTestCase';

export default class NumberTestCaseShadowDOM extends React.Component {
  componentWillMount() {
    // I tucked the definition in here because I wasn't sure what extending HTMLElement
    // would do in IE9. Maybe this isn't necessary.
    class ShadowNumberInput extends HTMLElement {
      connectedCallback(){
        ReactDOM.render(<NumberTestCase />, this.attachShadow({ mode: 'open' }));
      }
    }

    customElements.define('with-shadow-dom', ShadowNumberInput);
  }

  render() {
    return (<shadow-number-input {...this.props} />);
  }
}
```

Here's a working example: https://jsfiddle.net/2n77ootd/2/

Reply:
So whenever I go with that sort of an implementation, I get the following error when it tries to render the shadow element:

`TypeError: Failed to construct 'HTMLElement': Please use the 'new' operator, this DOM object constructor cannot be called as a function.`

It seems like custom elements are [not compatible with ES5 style classes](https://github.com/webcomponents/custom-elements#es5-vs-es2015). I'm able to get it working with their suggested solution, [native-shim.js](https://github.com/webcomponents/custom-elements/blob/master/src/native-shim.js). Is this something we'd be able to include in react? Or should we be looking for a different solution to this?

Reply:
@nhunzaker thoughts on this?